src/modules/dpos/validator/components/DelegatesTable/index.js

```diff
@@ -1,54 +1,80 @@
 /* istanbul ignore file */
 import React, { useMemo } from 'react';
-import { filterDelegates } from '../../utils';
-import TableWrapper from './TableWrapper';
+import { useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { QueryTable } from 'src/theme/QueryTable';
+import { useSort } from 'src/modules/common/hooks';
+import { useDelegates } from '../../hooks/queries';
+import { useForgersGenerator } from '../../hooks/queries/useForgersGenerator';
+import DelegateRow from './delegateRow';
+import header from './tableHeader';
+import { ROUND_LENGTH } from '../../consts';
 
 const DelegatesTable = ({
   setActiveTab,
-  delegates,
-  watchList,
-  hasLoadMore,
   activeTab,
-  changeSort,
   blocks,
   filters,
-  sort,
 }) => {
-  const { delegatesFilter, canLoadMore, currentOffset } = useMemo(() => {
-    const { offset: defaultOffset, count, total } = delegates?.meta || {};
-    const offset = defaultOffset + count;
-    return {
-      delegatesFilter: filterDelegates(delegates, filters),
-      canLoadMore: hasLoadMore && (offset < total),
-      currentOffset: offset,
-    };
-  }, [delegates, filters]);
+  const { t } = useTranslation();
+  const watchList = useSelector((state) => state.watchList);
+  const queryHook = activeTab === 'active' ? useForgersGenerator : useDelegates;
+  const { sort, toggleSort } = useSort();
 
-  const handleLoadMore = () => {
-    delegatesFilter.loadData(
-      Object.keys(filters).reduce(
-        (acc, key) => ({
-          ...acc,
-          ...(filters[key] && { [key]: filters[key] }),
-        }),
-        {
-          offset: currentOffset,
-        },
-      ),
-    );
-  };
+  // eslint-disable-next-line max-statements
+  const queryConfig = useMemo(() => {
+    const result = { config: { params: { ...filters, ...sort && { sort } } } };
+
+    if (activeTab === 'standby') {
+      result.config.params = {
+        ...result.config.params,
+        status: 'standby,ineligible',
+      };
+      return result;
+    }
+    if (activeTab === 'active') {
+      result.config.params = {
+        ...result.config.params,
+        limit: ROUND_LENGTH,
+      };
+      return result;
+    }
+    if (activeTab === 'sanctioned') {
+      result.config.params = {
+        ...result.config.params,
+        status: 'punished,banned',
+      };
+      return result;
+    }
+    if (activeTab === 'watched') {
+      result.config.params = {
+        ...result.config.params,
+        addressList: watchList,
+      };
+      return result;
+    }
+
+    return result;
```

Comment:
option 1
```suggestion
const params =  { 
     ...filters, 
     ...sort && { sort } ,
     ....(activeTab === 'standby' && { status: 'standby,ineligible' }),
     ....(activeTab === 'active' && { limit: ROUND_LENGTH }),
     ....(activeTab === 'sanctioned' && { status: 'punished,banned' }),
     ....(activeTab === 'watched' && { addressList: watchList }),
     }
     
    return {  config: { params } };
```

option 2
```suggestion
function getParams () {
     if(activeTab === 'standby')  return { status: 'standby,ineligible' });  
     if(activeTab === 'active')  return { limit: ROUND_LENGTH });  
     if(activeTab === 'sanctioned')  return { status: 'punished,banned' });  
     f(activeTab === 'watched')  return { addressList: watchList });  
}
     
    return {  config: { params: { 
     ...filters, 
     ...sort && { sort } ,
     ...getParams()
     } } };
```

Reply:
done